File: poc-ddd/src/domain/product-module/factory/product.factory.ts
import Product from "../entity/product";

export default class ProductFactory {

    public static create(name: string, price: number, type: string): Product {
        switch (type) {
            case '1':
                return new Product(name, price);
            default:
                throw new Error('Invalid type when create product.');
        }
    }

}

File: poc-arq-hexagonal/src/main/java/aristides/dev/portsadapters/product/core/factory/ProductFactory.java
package aristides.dev.portsadapters.product.core.factory;

import aristides.dev.portsadapters.product.core.entity.Product;

import java.math.BigDecimal;

public class ProductFactory {
    private ProductFactory() {
    }

    public static Product create(String name, BigDecimal price) {
        return new Product(name, price);
    }

}
